pallets/funding/src/functions.rs

```diff
@@ -2075,12 +2042,16 @@ impl<T: Config> Pallet<T> {
 					.checked_mul_int(usd_bond_needed)
 					.ok_or(Error::<T>::BadMath)?;
 
-				T::NativeCurrency::release(
-					&HoldReason::Participation(project_id).into(),
-					&bid.bidder,
-					bid.plmc_bond.saturating_sub(plmc_bond_needed),
-					Precision::Exact,
-				)?;
+				let plmc_bond_returned = bid.plmc_bond.saturating_sub(plmc_bond_needed);
+				// If the free balance of a user is zero and we want to send him less than ED, it will fail.
+				if plmc_bond_returned > T::ExistentialDeposit::get() {
+					T::NativeCurrency::release(
+						&HoldReason::Participation(project_id).into(),
+						&bid.bidder,
+						plmc_bond_returned,
+						Precision::Exact,
+					)?;
+				}
```

Comment:
![CleanShot 2024-04-19 at 16.00.55.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/Rk8O2JJI4xgR8dtiq7Dh/c9a4a37d-09fa-419b-b20e-6309ccf9e66a.png)

This is the logic for placing a hold. It is called with `Protect` and `Force`.
The case for erroring out when there's no ED left after hold is when there is only 1 provider. And if I understand correctly, we could in the future introduce pallets that provide for the existence of the account, making it possible for it to live without an ED.

Therefore I prefer to be safe and do a check before releasing so we don't halt the whole project. I don't think most people will be getting a return below ED anyway, and if they do they shouldn't mind it not being returned.

@lrazovic any thoughts?

Reply:
![CleanShot 2024-04-19 at 16.04.25.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/Rk8O2JJI4xgR8dtiq7Dh/f6f3b283-0bcd-40d1-915e-b3ff4f16fd96.png)

For reference here is the logic for releasing an asset. It always checks the amount free + the released amount will be bigger than ED.

Reply:
If it fails due to ED, we can cover it. If it fails for some other reason we probably don't want to continue the project transition because it's unexpected behavior and we should fix it and then force run the community start.

So I think we can resolve as is